bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java

```diff
@@ -0,0 +1,521 @@
+        }
+    }
+
+    private void updatePrices() {
+        cacheManager.cleanup();
+
+        updateCurrentSpotPrice();
+        updateCurrentTariff(CHANNEL_NET_TARIFF, cacheManager.getNetTariff());
+        updateCurrentTariff(CHANNEL_SYSTEM_TARIFF, cacheManager.getSystemTariff());
+        updateCurrentTariff(CHANNEL_ELECTRICITY_TAX, cacheManager.getElectricityTax());
+        updateCurrentTariff(CHANNEL_TRANSMISSION_NET_TARIFF, cacheManager.getTransmissionNetTariff());
+        updateHourlyPrices();
+
+        reschedulePriceUpdateJob();
+    }
+
+    private void updateCurrentSpotPrice() {
+        if (!isLinked(CHANNEL_SPOT_PRICE)) {
+            return;
+        }
+        BigDecimal spotPrice = cacheManager.getSpotPrice();
+        updateState(CHANNEL_SPOT_PRICE, spotPrice != null ? new DecimalType(spotPrice) : UnDefType.UNDEF);
+    }
+
+    private void updateCurrentTariff(String channelId, @Nullable BigDecimal tariff) {
+        if (!isLinked(channelId)) {
+            return;
+        }
+        updateState(channelId, tariff != null ? new DecimalType(tariff) : UnDefType.UNDEF);
+    }
+
+    private void updateHourlyPrices() {
+        if (!isLinked(CHANNEL_HOURLY_PRICES)) {
+            return;
+        }
+        Map<Instant, BigDecimal> spotPriceMap = cacheManager.getSpotPrices();
+        Price[] targetPrices = new Price[spotPriceMap.size()];
+        List<Entry<Instant, BigDecimal>> sourcePrices = spotPriceMap.entrySet().stream()
+                .sorted(Map.Entry.<Instant, BigDecimal> comparingByKey()).toList();
+
+        int i = 0;
+        for (Entry<Instant, BigDecimal> sourcePrice : sourcePrices) {
+            Instant hourStart = sourcePrice.getKey();
+            BigDecimal netTariff = cacheManager.getNetTariff(hourStart);
+            BigDecimal systemTariff = cacheManager.getSystemTariff(hourStart);
+            BigDecimal electricityTax = cacheManager.getElectricityTax(hourStart);
+            BigDecimal transmissionNetTariff = cacheManager.getTransmissionNetTariff(hourStart);
+            targetPrices[i++] = new Price(hourStart.toString(), sourcePrice.getValue(), config.currencyCode, netTariff,
+                    systemTariff, electricityTax, transmissionNetTariff);
+        }
+        updateState(CHANNEL_HOURLY_PRICES, new StringType(gson.toJson(targetPrices)));
+    }
+
+    /**
+     * Get the configured {@link Currency} for spot prices.
+     * 
+     * @return Spot price currency
+     */
+    public Currency getCurrency() {
+        return config.getCurrency();
+    }
+
+    /**
+     * Get cached spot prices or try once to download them if not cached
+     * (usually if no items are linked).
+     *
+     * @return Map of future spot prices
+     */
+    public Map<Instant, BigDecimal> getSpotPrices() {
+        try {
+            downloadSpotPrices();
+        } catch (DataServiceException e) {
+            logger.warn("Error retrieving spot prices");
```

Comment:
Log the exception's message? Same for below.

Reply:
Good idea, I must have forgotten this.